File: server/BidMasterOnline/BidMasterOnline.Domain/Entities/Complaint.cs
using BidMasterOnline.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace BidMasterOnline.Domain.Entities
{
    public class Complaint : EntityBase
    {
        public long AccusingUserId { get; set; }

        public long? AccusedUserId { get; set; }

        public long? AccusedAuctionId { get; set; }

        public long? AccusedCommentId { get; set; }

        public long? ModeratorId { get; set; }

        [MaxLength(5000)]
        public required string ComplaintText { get; set; }

        public ComplaintType Type { get; set; }

        public ComplaintStatus Status { get; set; }

        public User? Moderator { get; set; }

        public User? AccusingUser { get; set; }

        public User? AccusedUser { get; set; }

        public Auction? AccusedAuction { get; set; }

        public AuctionComment? AccusedComment { get; set; }
    }
}

File: client/ModeratorUI/src/app/models/complaints/complaint.ts
import { AuctionComment } from "../auctions/auctionComment";
import { BaseEntityModel } from "../baseEntityModel";
import { UserFeedback } from "../users/userFeedback";
import { ComplaintStatusEnum } from "./complaintStatusEnum";
import { ComplaintTypeEnum } from "./complaintTypeEnum";

export class Complaint extends BaseEntityModel {
  public accusingUserId!: number;
  public accusedUserId!: number;
  public accusedAuctionId?: number | null;
  public accusedCommentId?: number | null;
  public accusedUserFeedbackId?: number | null;
  public moderatorId?: number | null;
  public title!: string;
  public complaintText!: string;
  public type!: ComplaintTypeEnum;
  public status!: ComplaintStatusEnum;
  public moderatorConclusion?: string | null;
  public moderatorName?: string | null;
  public accusingUserName!: string;
  public accusedUsername!: string;
  public accusedAuctionName?: string;
  public accusedComment?: AuctionComment | null;
  public accusedUserFeedback?: UserFeedback | null;
}
